src/registries/npm-registry.js

```diff
@@ -133,10 +113,8 @@ export default class NpmRegistry extends Registry {
       opts.headers,
     );
 
-    const isToRegistry = this.isRequestToRegistry(requestUrl, registry);
-
     // this.token must be checked to account for publish requests on non-scopped packages
-    if (this.token || (isToRegistry && (alwaysAuth || this.isScopedPackage(packageIdent)))) {
+    if (this.token || alwaysAuth || this.isScopedPackage(packageIdent)) {
```

Comment:
The key change is basically:

We stop checking if request isToRegistry (which is a strange and slightly complicated thing to understand, of course request is being made to a registry, what else would it be made to?)

And instead we make use of `this.getAuth` to figure out if this specific request needs a token or not.

In fact, we could push these 3 conditions (`this.token`, `alwaysAuth`, `scopedPackage`) to `getAuth` making this method much cleaner. `getAuth` takes a url and figures out if authorization is required or not. Should I do that?